src/Promact.TestCaseManagement.Core/Controllers/ModuleController.cs

```diff
@@ -0,0 +1,157 @@
+                return NotFound();
+            }
+
+            var module = await _iModuleRepository.GetModuleAsync(projectId, id);
+
+            if (module == null)
+            {
+                return NotFound();
+            }
+
+            return Ok(module);
+        }
+
+        /// <summary>
+        /// Method to add module to the database
+        /// </summary>
+        /// <param name="projectId">Id of the project</param>
+        /// <param name="moduleAC">Module object</param>
+        /// <returns></returns>
+        [HttpPost("{projectId}/module")]
+        public async Task<IActionResult> CreateModuleAsync(int projectId, [FromBody]ModuleAC moduleAC)
+        {
+            if (!ModelState.IsValid)
+            {
+                return BadRequest(ModelState);
+            }
+
+            if (!await _iProjectRepository.IsProjectExistAsync(projectId))
+            {
+                return NotFound();
+            }
+            var module = Mapper.Map<Module>(moduleAC);
+            module.ProjectId = projectId;
+
+            return Ok(await _iModuleRepository.AddModuleAsync(module));
+        }
+
+        /// <summary>
+        /// Method to update module to the database
+        /// </summary>
+        /// <param name="projectId">Id of the project</param>
+        /// <param name="id">Id of the module</param>
+        /// <param name="moduleAC">Module object</param>
+        /// <returns></returns>
+        [HttpPut("{projectId}/module/{id}")]
+        public async Task<IActionResult> UpdateModuleAsync(int projectId, int id, [FromBody] ModuleAC moduleAC)
+        {
+            if (!ModelState.IsValid)
+            {
+                return BadRequest(ModelState);
+            }
+
+            if (!await _iProjectRepository.IsProjectExistAsync(projectId))
+            {
+                return NotFound();
+            }
+            var module = await _iModuleRepository.GetModuleAsync(projectId, id);
+            if (module == null)
+            {
+                return NotFound();
+            }
+
+            Mapper.Map(moduleAC, module);
+            await _iModuleRepository.UpdateModuleAsync(module);
+
+            return NoContent();
+        }
+
+        /// <summary>
+        /// Method to delete module from the database
+        /// </summary>
+        /// <param name="projectId">Id of the project</param>
+        /// <param name="id">Id of the module</param>
+        /// <returns></returns>
+        [HttpDelete("{projectId}/module/{id}")]
+        public async Task<IActionResult> DeleteModule(int projectId, int id)
+        {
+            if (!await _iProjectRepository.IsProjectExistAsync(projectId))
+            {
+                return NotFound();
+            }
+
+            var module = await _iModuleRepository.GetModuleAsync(projectId, id);
+
+            if (module == null)
+            {
+                return NotFound();
+            }
+
+            await _iModuleRepository.DeleteModuleAsync(module);
```

Comment:
In case there is a DB exception, you should return 400 not 404 or 500. because the data that is sent are invalid which will cause server to respond that it was a bad request.